dftimewolf/cli/curses_display_manager.py

```diff
@@ -0,0 +1,290 @@
+  PENDING = 'Pending'
+  CANCELLED = 'Cancelled'
+
+
+class Module:
+  """An object used by the CursesDisplayManager used to represent a DFTW module.
+  """
+  def __init__(self,
+               name: str,
+               dependencies: List[str],
+               runtime_name: Optional[str] = None):
+    """Initialise the Module object."""
+    self.name = name
+    self.runtime_name = runtime_name if runtime_name else name
+    self.status: Status = Status.PENDING
+    self._dependencies: List[str] = dependencies
+    self._error_message: str = ''
+    self._threads: Dict[str, Dict[str, Any]] = {}
+    self._threads_containers_max: int = 0
+    self._threads_containers_completed: int = 0
+
+  def Stringify(self) -> List[str]:
+    """Returns an CursesDisplayManager friendly string describing the module."""
+    module_line = f'     {self.runtime_name}: {self.status.value}'
+    thread_lines = []
+
+    if self.status == Status.PENDING and len(self._dependencies) != 0:
+      module_line += f' ({", ".join(self._dependencies)})'
+    elif self.status == Status.ERROR:
+      module_line += f': {self._error_message}'
+    elif self.status in [Status.RUNNING, Status.PROCESSING] and self._threads:
+      module_line += (f' - {self._threads_containers_completed} of '
+          f'{self._threads_containers_max} containers completed')
+      for n, t in self._threads.items():
+        thread_lines.append(
+            f'       {n}: {t["status"].value} ({t["container"]})')
+
+    return [module_line] + thread_lines
+
+  def SetStatus(self, status: Status) -> None:
+    """Set the status of the module."""
+    if self.status not in [Status.ERROR, Status.COMPLETED, Status.CANCELLED]:
+      self.status = status
+
+  def SetThreadState(self, thread: str, status: Status, container: str) -> None:
+    """Set the state of a thread within a threaded module."""
+    self._threads[thread] = {'status': status,
+                             'container': container}
+    if status == Status.COMPLETED:
+      self._threads_containers_completed += 1
+
+  def SetError(self, message: str) -> None:
+    """Sets the error for the module."""
+    self._error_message = message
+    self.status = Status.ERROR
+
+  def SetContainerCount(self, count: int) -> None:
+    """Sets the maximum container count for the module."""
+    self._threads_containers_max = count
+
+
+class Message:
+  """Helper class for managing messages."""
+
+  def __init__(self, source: str, content: str, is_error: bool = False) -> None:
+    """Initialise a Message object."""
+    self.source: str = source
+    self.content: str = content
+    self.is_error: bool = is_error
+
+  def Stringify(self, source_buff_len: int = 0, colourise: bool = False) -> str:
+    """Returns an CursesDisplayManager friendly string of the Message."""
+    pad = (len(self.source) if len(self.source) > source_buff_len
+        else source_buff_len)
+
+    colour_code = '\u001b[31m' if self.is_error and colourise else ''
```

Comment:
So far we've used americanized English, e.g. `color`. Would you mind adjusting this? Not a hill I'm willing to die on but I guess it's better to make it consistent throughout the codebase.

Reply:
As much as it pains me, done. I think I got it all, but since it's my native, I might have missed some.